components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -432,63 +442,197 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes) throw
         }
     }
 
-    private List<Object> filterUsers(Node node, Map<String, Boolean> requiredAttributes, int offset, int limit)
+    private List<Object> filterUsers(Node node, Map<String, Boolean> requiredAttributes, int offset, int limit,
+                                     String sortBy, String sortOrder, String domainName)
             throws NotImplementedException, CharonException {
 
-        if(node.getLeftNode() != null || node.getRightNode() != null){
-            String error = "Complex filters are not supported yet";
-            throw new NotImplementedException(error);
+        List<Object> filteredUsers = new ArrayList<>();
+        //0th index is to store total number of results
+        filteredUsers.add(0);
+        String[] userNames;
+
+        if (node instanceof ExpressionNode) {
+            String attributeName = ((ExpressionNode) node).getAttributeValue();
+            String filterOperation = ((ExpressionNode) node).getOperation();
+            String attributeValue = ((ExpressionNode) node).getValue();
+
+            if (log.isDebugEnabled()) {
+                log.debug("Listing users by filter: " + attributeName + filterOperation +
```

Comment:
Separate printing values, to make it understandable easily

Reply:
fixed